server/controllers/register.js

```diff
@@ -0,0 +1,31 @@
+const bcrypt = require('bcrypt');
+
+
+const {checkEmail, addUser} = require('../database/queries');
+const {regSchema} = require('../utils/validations')
+
+const signup = (req, res) => {
+    const user = req.body;
+
+    const userValid = regSchema.validateAsync(user);
+
+    const userExist = (rows) => {
+        if(rows.rowCount > 0){
+            throw new Error('user registered');
+        }
+
+    };
+
+    userValid
+    .then((user) => checkEmail(user.email))
+    .then(rows => userExist(rows))
+    .then(() => bcrypt.hash(user.password, 10))
+    .then((newPass) => addUser(user, newPass))
+    .then(()=> res.status(200).redirect('/login'))
```

Comment:
You shouldn't redirect to the login page in order to send cookies to the user after registration, you can use the `jwt.sign` by yourself or use it as a middleware that comes after the signup controller